test/puppetlabs/puppetdb/threadpool_test.clj

```diff
@@ -32,14 +32,39 @@
 
       (with-open [gtp (gated-threadpool 1 "test-pool-%d" 1000)]
         (let [fut (future
-                    (dochan gtp worker-fn in-chan))]
+                    (dochan gtp worker-fn in-chan tu/blocked?))]
           (async/>!! in-chan {:id :a})
           (is (= [:done :a] (async/<!! out-chan)))
           (is (= 1 @counter))
           (async/close! in-chan)
           (is (not= ::timed-out (deref fut tu/default-timeout-ms ::timed-out)))
           (is (true? (future-done? fut)))))))
 
+
+  (testing "block command"
+    (let [counter (atom 0)
+          in-chan (async/chan 10)
+          out-chan (async/chan 10)
+          worker-fn (wrap-out-chan
+                      out-chan
+                      (fn [{:keys [id]}]
+                        (swap! counter inc)
+                        [:done id]))
+          blocked? (atom true)]
+
+        (with-open [gtp (gated-threadpool 1 "test-pool-%d" 1000)]
+          (let [fut (future
+                      (dochan gtp worker-fn in-chan blocked?))]
+            (async/>!! in-chan {:id :a})
+            (let [channel (future (async/<!! out-chan))
+                  ]
+              (is (= (deref channel 1000 "Did not wake up!") "Did not wake up!") )
```

Comment:
Extra space at the end of the line, and also suggest putting the expected value first in `is` tests, i.e. `(is (= EXPECTED ACTUAL))`.  I believe that's more idiomatic, and I *think* some tooling might expect it (when rendering results, etc.).

Reply:
changed